### ISSUE TITLE
Guards of decision nodes correctly evaluated ?

### ISSUE THREAD

**Issue by oppm** [pnp-software/fwprofile#19]
I am not totally sure if this is a bug, but I cannot explain the behaviour otherwise. 
- I used the provided Example 3 project and downloaded the code to build an executable. I therefore had to provide an implementation for Guard1 and Guard2 of the decision node. 
- In Guard1 I return false (0) and Guard2 I return true (1). When running the procedure, I assumed that "Node 4: Action 2" will be executed. 
- However, it is not. Instead, Guard1 is always evaluated over and over again. This is wrong, isn't it ? 

I had a quick look at the code, but find it difficult to understand. However, line 81 in FwPrCore.c seems a bit odd as the comment for the loop indicates "Execute loop **until** guard ... is true" while the (pseudo-) code is "while (trueGuardFound == true) {"...

**Comment by pasetti** [comment]
The segment of code around line 81 in FwPrCore.c is new and was introduced in March. Hence, it is possible that there is an error here.

What do you mean by "Example 3"? We only have one single example for the procedure module. Can you send me the example code (or point me to its location).

The comment above the while loop is incorrect. It should read: "Execute loop as long as the guard of the control flow out of the current node is true".

**Comment by oppm** [comment]
I meant the standard examples (Presets) we provide in the framework editor. Example 1 and 2 are for state machines, while example 3 is for procedures. 
Anyway, stupid me. I didn't notice that the same Guard is included twice in the example. So when I locked the guard from the decision node, I actually also locked the guard somewhere else. That's why it was stuck at the end. So, no bug.
I will update the above mentioned comment anyway.
